weldx/asdf/tags/weldx/core/dataset.py

```diff
@@ -80,17 +72,9 @@ def from_tree(cls, tree, ctx):
             An instance of the 'xarray.Dataset' type.
 
         """
-        data_vars = {}
-
-        for variable in tree["variables"]:
-            data_vars[variable.name] = (variable.dimensions, variable.data)
-
-        coords = {}
-        for coordinate in tree["coordinates"]:
-            coords[coordinate.name] = (coordinate.dimensions, coordinate.data)
-
-        obj = Dataset(data_vars=data_vars, coords=coords)
+        data_vars = {v.name: (v.dimensions, v.data, v.attrs) for v in tree["variables"]}
+        coords = {c.name: (c.dimensions, c.data, c.attrs) for c in tree["coordinates"]}
 
-        obj.attrs = tree["attributes"]
+        ds = Dataset(data_vars=data_vars, coords=coords, attrs=tree["attributes"])
 
-        return obj
+        return ds
```

Comment:
```suggestion
```

Reply:
Sorry, for the messed-up suggestions, but GitHub seems to have problems with suggestions that have some unchanged or removed lines in between. You probably need to reformat the code afterward.